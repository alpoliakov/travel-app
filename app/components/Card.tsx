import { useRouter } from 'next/router';
import React from 'react';

import { useAuth } from '../lib/useAuth';

export default function Card({ item }) {
  const { _id, data, imagesUrl } = item;
  const { user } = useAuth();

  const router = useRouter();
  const { locale } = useRouter();

  const goToCountryPage = (e) => {
    const currentId = e.target.dataset.id;
    console.log(currentId);
    if (!user) {
      return;
    }

    router.push(`/countries/${_id}`);
  };

  return (
    <div
      className="card hover__item shadow__item p-3"
      role="presentation"
      onClick={goToCountryPage}
      data-id={_id}
      style={{
        backgroundImage: `url(${imagesUrl[0]}), linear-gradient(var(--color-grey-light) 1%, var(--color-grey-dark) 100%)`,
      }}>
      <h1
        className="shadow__item gradient-text font-semibold"
        data-id={_id}
        style={{ fontSize: 42, zIndex: -1 }}>
        {data[locale].name}
      </h1>
      <h1
        className="shadow__item lowercase gradient-text font-semibold text-2xl"
        data-id={_id}
        style={{ color: '#ffffff', zIndex: -1 }}>
        {data[locale].capital}
      </h1>
    </div>
  );
}
