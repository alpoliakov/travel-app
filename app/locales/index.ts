export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    forgetPass: 'Forgot password ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    forgetPass: 'Забыли пароль ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    forgetPass: 'Забули пароль ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
  },
};
