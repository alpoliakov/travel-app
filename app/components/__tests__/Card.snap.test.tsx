import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { messages } from '../../locales';
import Card from '../Card';

const item = {
  __typename: 'Country',
  _id: '60450fb4d4f7b1666c3bf94b',
  ISOCode: 'DEU',
  imagesUrl: [
    'https://images.unsplash.com/photo-1554072675-66db59dba46f?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1440&q=80'
  ],
  videoUrl: 'https://www.youtube.com/watch?v=IPbzWJNmndY',
  currency: 'EUR',
  timeZone: 'CET',
  coordinates: [52.31, 13.23],
  data: {
    __typename: 'LocaleCountry',
    en: {
      __typename: 'DataLocaleCountry',
      name: 'Germany',
      description:
        "'Germany is a land of diverse landscapes and interesting cities. Its economy is the largest in Europe and the fifth largest in the world. It was in Germany that many revolutionary scientific discoveries were made: science and research have a long tradition here.'",
      capital: 'Berlin',
    },
    ru: {
      __typename: 'DataLocaleCountry',
      name: 'Германия',
      description:
        'Германия - страна разнообразных пейзажей и интересных городов. Ее экономика является крупнейшей в Европе и пятой по величине в мире. Именно в Германии были сделаны многие революционные научные открытия: наука и исследования имеют здесь давние традиции.',
      capital: 'Берлин',
    },
    uk: {
      __typename: 'DataLocaleCountry',
      name: 'Німеччина',
      description:
        "Німеччина - країна різноманітних пейзажів і цікавих міст. Її економіка є найбільшою в Європі і п'ятою за величиною в світі. Саме в Німеччині були зроблені багато революційні наукові відкриття: наука і дослідження мають тут давні традиції.",
      capital: 'Берлін',
    },
  },
};


jest.mock('../../lib/useAuth', () => ({
  useAuth: jest.fn(() => ({ signIn: jest.fn(), user: '', error: '' })),
  AuthProvider: ({ children }) => <div>{children}</div>,
}));

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    route: '/',
    pathname: '',
    query: '',
    asPath: '',
    locale: 'en',
    locales: ['en', 'ru', 'uk'],
  }),
}));

it('Card renders correctly', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="ru" messages={messages['ru']}>
        <Card item={item} />)
      </IntlProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
