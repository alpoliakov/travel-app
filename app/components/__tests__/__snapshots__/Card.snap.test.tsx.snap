// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
Array [
  <div
    className="card hover__item shadow__item p-3"
    data-id="60450fb4d4f7b1666c3bf94b"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "backgroundImage": "url(https://images.unsplash.com/photo-1554072675-66db59dba46f?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1440&q=80), linear-gradient(var(--color-grey-light) 1%, var(--color-grey-dark) 100%)",
      }
    }
  >
    <h1
      className="shadow__item gradient-text text-2xl sm:text-3xl md:text-4xl font-semibold"
      data-id="60450fb4d4f7b1666c3bf94b"
      style={
        Object {
          "zIndex": -1,
        }
      }
    >
      Germany
    </h1>
    <h1
      className="shadow__item lowercase gradient-text text-xl font-semibold md:text-2xl"
      data-id="60450fb4d4f7b1666c3bf94b"
      style={
        Object {
          "color": "#ffffff",
          "zIndex": -1,
        }
      }
    >
      Berlin
    </h1>
  </div>,
  ")",
]
`;
